requests/exceptions.py

```diff
@@ -101,6 +101,8 @@ class RetryError(RequestException):
 class ConflictingHeaderError(RequestException):
     """Mutually exclusive request headers set"""
 
+class UnreachableCodeError(RequestException, RuntimeError):
+    """Unreachable code block reached"""
```

Comment:
Heh, while I like this name I don't think it's a good one. It's too general. =)

Let's instead define a new exception that is more like this:

```python
class InvalidBodyError(RequestException, ValueError):
    """An invalid request body was specified."""
```